### ISSUE TITLE
version GLIBC_2.29 not found

### ISSUE THREAD

**Issue by dr-death2** [aclap-dev/vdhcoapp#135]
I'm running VdhCoApp 1.6.3 with FF 102.0.1 on Linux Mint 19.3.

VdhCoApp used to work for many years in the past on the same system, but downloads now seem to be interrupted when the download is at 100%. It fails with

```
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libavdevice.so.58)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libavfilter.so.7)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libavformat.so.58)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libavcodec.so.58)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libswresample.so.3)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libswscale.so.5)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libavutil.so.56)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libavutil.so.56)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libvpx.so.6)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libwebpmux.so.3)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libwebp.so.7)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libaom.so.0)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libmp3lame.so.0)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libopenjp2.so.7)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libopus.so.0)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libvorbis.so.0)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libx264.so.161)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libx265.so.188)
/opt/net.downloadhelper.coapp/converter/build/linux/64/ffmpeg: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /opt/net.downloadhelper.coapp/converter/build/linux/64/libxvidcore.so.4)
```

So downloading is impossible ATM.

Searching the web brought me to https://groups.google.com/g/video-downloadhelper-q-and-a/c/9oTmcvGFHy0 and https://groups.google.com/g/video-downloadhelper-q-and-a/c/zF42AmjVTQU/m/FoF6UCLrAwAJ. That's why i expect the cause to be a compiling issue on Ubuntu unintentionally excluding other Linux systems.

```
ldd --version

ldd (Ubuntu GLIBC 2.27-3ubuntu1.6) 2.27
Copyright © 2018 Free Software Foundation, Inc.
Dies ist freie Software; in den Quellen befinden sich die Lizenzbedingungen.
Es gibt KEINERLEI Garantie; nicht einmal für die TAUGLICHKEIT oder
VERWENDBARKEIT FÜR EINEN ANGEGEBENEN ZWECK.
Implementiert von Roland McGrath und Ulrich Drepper.
```

**Comment by mi-g** [comment]
Each release of the vdh coapp comes with several flavours of glibc versions ([https://github.com/mi-g/vdhcoapp/releases/tag/v1.6.3](https://github.com/mi-g/vdhcoapp/releases/tag/v1.6.3)) to accommodate various linux distros. Did you try other `.tar.gz` packages ?

**Comment by dr-death2** [comment]
Thanks for your reply. I downloaded the one suggested at https://www.downloadhelper.net/install-coapp for "Linux - 64 bits - deb". This links to `https://github.com/mi-g/vdhcoapp/releases/download/v1.6.3/net.downloadhelper.coapp-1.6.3-1_amd64.deb` and says nothing about possible conditions or caveats.

That setup worked in the past until some unknown circumstances occured. It might have been an Kernel update, FF update or VdhCoApp update or something else.

With your hint I was able to fix my problem and this is what i did:

1. uninstall non functional version
`sudo dpkg -r "net.downloadhelper.coapp"`

2.  download the version that does **not** require the latest GLIBC version from `https://github.com/mi-g/vdhcoapp/releases/tag/v1.6.3`
in my case i used `https://github.com/mi-g/vdhcoapp/releases/download/v1.6.3/net.downloadhelper.coapp-1.6.3-1_amd64-glibc-2.27.deb`

3. install the downloaded version
`sudo dpkg -i "net.downloadhelper.coapp-1.6.3-1_amd64-glibc-2.27.deb"`

That fixed my problem.

But what makes me wonder is that conditions exist or can occur, that render (just only!) VdhCoApp useless. In case it was a client system update all other software stayed usable. So i still think that something is wrong with your compiling setup or packaging. Maybe your way of linking to a specific `glibc` is not  the best choice. If you feel the need of providing multiple VdhCoApp "versions" for each possible `glibc` version then your build system is not ideal. Not to mention that the way of fixing the issue is not very user friendly and is beyond most users capabilities.
